private/pkg/tagrangeutil/tagrangeutil_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tagrangeutil
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/bufbuild/buf/private/pkg/protosource"
+	"github.com/stretchr/testify/assert"
+)
+
+// Implementation of protosource.TagRange for testing.
+type testTagRange struct {
+	start, end int
+}
+
+func (testTagRange) File() protosource.File { return nil }
+
+func (testTagRange) Location() protosource.Location { return nil }
+
+func (r testTagRange) Start() int { return r.start }
+
+func (r testTagRange) End() int { return r.end }
+
+func (testTagRange) Max() bool { return false }
+
+// Parses a string like 5-10,12 into test ranges.
+func stringToTestRanges(rangesString string) []protosource.TagRange {
+	results := []protosource.TagRange{}
+	if rangesString == "" {
+		return results
+	}
+	rangeStrings := strings.Split(rangesString, ",")
+	for _, rangeString := range rangeStrings {
+		beginString, endString, hasEnd := strings.Cut(rangeString, "-")
+		if !hasEnd {
+			endString = beginString
+		}
+		begin, err := strconv.Atoi(beginString)
+		if err != nil {
+			panic(err.Error())
```

Comment:
nit: `panic(err)` works :)

Reply:
Fixed. For some reason, all of the existing instances of `panic` are doing this, so I mistook it for a stylistic decision.